gcloud/search/index.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Indexes."""
+
+
+class Index(object):
+    """Indexes are containers for documents.
+
+    See:
+    https://cloud.google.com/search/reference/rest/v1/indexes
+
+    :type name: string
+    :param name: the name of the index
+
+    :type client: :class:`gcloud.dns.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the index (which requires a project).
+    """
+
+    def __init__(self, name, client):
+        self.name = name
+        self._client = client
+        self._properties = {}
+
+    @classmethod
+    def from_api_repr(cls, resource, client):
+        """Factory:  construct an index given its API representation
+
+        :type resource: dict
+        :param resource: index resource representation returned from the API
+
+        :type client: :class:`gcloud.dns.client.Client`
+        :param client: Client which holds credentials and project
+                       configuration for the index.
+
+        :rtype: :class:`gcloud.dns.index.Index`
+        :returns: Index parsed from ``resource``.
+        """
+        name = resource.get('indexId')
+        if name is None is None:
+            raise KeyError(
+                'Resource lacks required identity information: ["indexId"]')
```

Comment:
This is exactly the same thing we do for the DNS / PubSub `from_api_repr` methods.